src/L1/rollup/ScrollChain.sol

```diff
@@ -154,6 +160,23 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
         bytes32 withdrawRoot;
     }
 
+    /// @dev Assume one message is enqueued at timestamp `t`. The `maxInclusionDelay` means if this
+    /// message wasn't included by timestamp `t+maxInclusionDelay` in batch committing, the sequencer
+    /// cannot commit any batches unless it includes all expired messages before any L2 transactions.
+    /// The `maxFinalizeDelay` means if this message wasn't finalized by timestamp `t+maxFinalizeDelay`,
+    /// the prover cannot finalize any batches. The `maxDelayEnterEnforcedMode` means if this message
+    /// wasn't finalized by timestamp `t+maxDelayEnterEnforcedMode`, we will enter enforced mode. Anyone
+    /// can commit and finalize a batch without permission.
+    /// So if the sequencer or prover encountered some recoverable problem, we must resolve it between
+    /// timestamp `[t+maxFinalizeDelay, t+maxDelayEnterEnforcedMode]`.
+    struct EnforcedBatchParameters {
+        uint64 maxInclusionDelay;
+        uint64 maxFinalizeDelay;
```

Comment:
how to resolve the issue when prover cannot finalize any batches before maxFinalizeDelay? Since `finalizeBundleWithProof` and `finalizeBundleWithTeeProof` are blocked by:
```
_checkFinalizationAllowed(enforcedBatchParameters.maxFinalizeDelay);
```